lib/src/stacked_table.rs

```diff
@@ -409,7 +409,9 @@ impl TableStore {
         let table = mut_table.save_in(self)?;
         self.add_head(&table)?;
         if let Some(parent_table) = maybe_parent_table {
-            self.remove_head(&parent_table);
+            if parent_table.name != table.name {
+                self.remove_head(&parent_table);
+            }
```

Comment:
I was going to ask if we could skip the rest of the function body as well if `mut_table` is empty, but I we still need to handle write of an initial empty table, so it's easier this way.